### PR TITLE
Clarify a warning

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/Prefix.jl [hunk 1]
```diff
@@ -431,7 +431,7 @@ function setup_dependencies(prefix::Prefix, dependencies::Vector{PkgSpec}, platf
             # Get the path to the main artifact
             artifact_hash = Pkg.Artifacts.artifact_hash(name[1:end-4], artifacts_toml; platform=platform)
             if artifact_hash === nothing
-                @warn("Dependency $(name) does not have a mapping for $(name[1:end-4])!")
+                @warn("Dependency $(name) does not have a mapping for artifact $(name[1:end-4])!")
```
Comment (fingolfin) [inline src/Prefix.jl:434]: See https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/854. I find this slightly more helpful... but perhaps an larger change would be even better... E.g. like this:
```suggestion
                @warn("Dependency $(name) does not have a mapping for artifact $(name[1:end-4]); perhaps it was not built for platform $(platform)")
```
Thoughts?

File: test/dependencies.jl [hunk 2]
```diff
@@ -69,7 +69,7 @@ end
                 Dependency("LibOSXUnwind_jll")
             ]
             platform = Linux(:i686, libc=:musl)
-            @test_logs (:warn, "Dependency LibOSXUnwind_jll does not have a mapping for LibOSXUnwind!") setup_dependencies(prefix, getpkg.(dependencies), platform)
+            @test_logs (:warn, "Dependency LibOSXUnwind_jll does not have a mapping for artifact LibOSXUnwind!") setup_dependencies(prefix, getpkg.(dependencies), platform)
```
Comment (fingolfin) [inline test/dependencies.jl:72]: The test will need to be adapted, too.
```suggestion
            @test_logs (:warn, "Dependency LibOSXUnwind_jll does not have a mapping for artifact LibOSXUnwind; perhaps it was not built for platform Linux(:i686, libc=:musl)") setup_dependencies(prefix, getpkg.(dependencies), platform)
```

File: src/Prefix.jl [hunk 3]
```diff
@@ -431,7 +431,7 @@ function setup_dependencies(prefix::Prefix, dependencies::Vector{PkgSpec}, platf
             # Get the path to the main artifact
             artifact_hash = Pkg.Artifacts.artifact_hash(name[1:end-4], artifacts_toml; platform=platform)
             if artifact_hash === nothing
-                @warn("Dependency $(name) does not have a mapping for $(name[1:end-4])!")
+                @warn("Dependency $(name) does not have a mapping for artifact $(name[1:end-4]); perhaps it was not built for platform $(platform)")
```
Comment (giordano) [inline src/Prefix.jl:434]: What about something like
```suggestion
                @warn("Dependency $(name) does not have a mapping for artifact $(name[1:end-4]) for platform $(platform)")
```
?

File: test/dependencies.jl [hunk 4]
```diff
@@ -69,7 +69,7 @@ end
                 Dependency("LibOSXUnwind_jll")
             ]
             platform = Linux(:i686, libc=:musl)
-            @test_logs (:warn, "Dependency LibOSXUnwind_jll does not have a mapping for LibOSXUnwind!") setup_dependencies(prefix, getpkg.(dependencies), platform)
+            @test_logs (:warn, "Dependency LibOSXUnwind_jll does not have a mapping for artifact LibOSXUnwind; perhaps it was not built for platform Linux(:i686, libc=:musl)") setup_dependencies(prefix, getpkg.(dependencies), platform)
```
Comment (giordano) [inline test/dependencies.jl:72]:
```suggestion
            @test_logs (:warn, "Dependency LibOSXUnwind_jll does not have a mapping for artifact LibOSXUnwind for platform Linux(:i686, libc=:musl)") setup_dependencies(prefix, getpkg.(dependencies), platform)
```

**Pull request merged**